feg/gateway/services/session_proxy/credit_control/model_conversion.go

```diff
@@ -45,6 +46,17 @@ func getCreditUnit(volume *uint64) *protos.CreditUnit {
 	return &protos.CreditUnit{IsValid: true, Volume: *volume}
 }
 
+func GetVirtualApnRule(apnFilter, apnOverwrite string) *VirtualApnRule {
```

Comment:
This functions seems to be here because it is the only common file between Gx and Gy. But `model_conversion` file only includes conversion from protos to specific types. So I don't think this function belongs here. 

Maybe you can create another file at this level or extend mconfig.pb.go to add this method to `VirtualApnRule` type. But here, to keep the code easy to understand, duplication should be just fine. Also see this piece of code is just some error checks and simple object building.